common/telemetry/config.go

```diff
@@ -418,3 +421,7 @@ func (e *exporter) UnmarshalYAML(n *yaml.Node) error {
 	}
 	return obj.Spec.Decode(e.Spec)
 }
+
+func debugMode() bool {
+	return os.Getenv(debugModeEnvVar) == "true"
```

Comment:
There's no great answer to this perennial question, but just checking -- you don't want to support for example `"1"`? If there's precedent for boolean-valued env var flags in the codebase then I'm definitely happy to follow that precedent. Otherwise I guess I think "1" is the main one I'd add.

Reply:
I was following the other OTEL env vars such as `OTEL_EXPORTER_OTLP_TRACES_INSECURE=true`; see https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#boolean-value

Now that I looked at this, they suggest being case-insensitive, which I'll add.

Reply:
> I'm open to using 1, too, if that's more popular.

Let's hope the OTEL spec authors don't see this comment.

Seriously though, no strong opinion. I'm not sure exactly who that spec is intended to apply to. If we have other env vars in the service that people use `1` with then it could be confusing. But no need to change now from my POV.

Reply:
You could also use `strconv.ParseBool()` to accept both :)

Reply:
Didn't know about that one; yeah, why not.